internal/paascharm/gostructs.go

```diff
@@ -0,0 +1,142 @@
+package paascharm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"unicode"
+)
+
+const CommonPrefix = "APP_"
+
+// Map from integration key in charmcraft.yaml to Go name
+var IntegrationsToGoName = map[string]string{
+	"mongodb":    "MongoDB",
+	"mysql":      "MySQL",
+	"postgresql": "PostgreSQL",
+	"redis":      "Redis",
+	"s3":         "S3",
+	"saml":       "SAML",
+}
+
+// Map from database integration key in charmcraft.yaml to its prefix in env vars
+var DatabaseIntegrationNameToPrefix = map[string]string{
+	"mongodb":    CommonPrefix + "MONGODB_",
+	"mysql":      CommonPrefix + "MYSQL_",
+	"postgresql": CommonPrefix + "POSTGRESQL_",
+	"redis":      CommonPrefix + "REDIS_",
+}
+
+// Charmcraft config options types to Go Types
+var CharmcraftToGoTypes = map[string]string{
+	"bool":    "bool",
+	"boolean": "bool",
+	"float":   "float64",
+	"int":     "int",
+	"secret":  "string",
+	"string":  "string",
+}
+
+type GoStructsData struct {
+	PackageName             string
+	CommonPrefix            string
+	Options                 []Option
+	HasDatabaseIntegrations bool
+	Integrations            map[string]Integration
+}
+
+type Option struct {
+	GoVarName  string
+	GoVarType  string
+	EnvVarName string
+}
+
+type Integration struct {
+	Name           string
+	GoName         string
+	Interface      string
+	Optional       bool
+	IsDatabase     bool
+	DatabasePrefix string
+}
+
+// Create a GoStructsData with all the information needed to generate the Go file from the charmcraft.yaml parsed file
+func NewGoStructsData(packageName string, charmcraft CharmcraftYamlConfig) (goStructs GoStructsData, err error) {
+	goStructs = GoStructsData{
+		PackageName:  packageName,
+		CommonPrefix: CommonPrefix,
+	}
+
+	for key, value := range charmcraft.Config.Options {
+		varType, err := buildGoVarType(value)
+		if err != nil {
+			return goStructs, err
+		}
+		option := Option{
+			GoVarName:  buildGoVarName(key),
+			GoVarType:  varType,
+			EnvVarName: buildEnvVarName(key),
+		}
+		goStructs.Options = append(goStructs.Options, option)
+	}
+
+	goStructs.Integrations = make(map[string]Integration)
+	for key, value := range charmcraft.Requires {
+		integration := Integration{
+			Name:      key,
+			Interface: value.Interface,
+			Optional:  value.Optional,
+		}
+
+		if goName, ok := IntegrationsToGoName[key]; ok {
+			integration.GoName = goName
+			if databasePrefix, okDatabase := DatabaseIntegrationNameToPrefix[key]; okDatabase {
+				goStructs.HasDatabaseIntegrations = true
+				integration.IsDatabase = true
+				integration.DatabasePrefix = databasePrefix
+			}
+			goStructs.Integrations[key] = integration
+		} else {
```

Comment:
Go style: it's more typical Go style to test `if !ok` and do an early `continue` to keep the happy path at the top level. For example:

```go
goName, ok := IntegrationsToGoName[key]
if !ok {
    log.Printf("Skipping")
    continue
}
// normal code path
```

Reply:
sure! thanks for the tip